subgraphs/opyn-gamma/src/price.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  Bytes,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import { NetworkConfigs } from "../configurations/configure";
+import { Oracle } from "../generated/Controller/Oracle";
+import { Option, Token } from "../generated/schema";
+import { BIGDECIMAL_ZERO, INT_EIGHT } from "./common/constants";
+import { getOrCreateToken } from "./common/tokens";
+import { bigIntToBigDecimal } from "./common/utils/numbers";
+import { getUsdPricePerToken } from "./prices";
+
+export function getTokenPrice(event: ethereum.Event, token: Token): BigDecimal {
+  if (event.block.number == token.lastPriceBlockNumber!) {
+    return token.lastPriceUSD!;
+  }
+  const oracle = Oracle.bind(
+    NetworkConfigs.getOracleAddress(event.block.number.toI32())
+  );
+  const priceResult = oracle.try_getPrice(Address.fromBytes(token.id));
+  if (!priceResult.reverted) {
+    const price = bigIntToBigDecimal(priceResult.value, INT_EIGHT);
+    token.lastPriceBlockNumber = event.block.number;
+    token.lastPriceUSD = price;
+    token.save();
+    return price;
+  }
+  log.error("Failed to get price for asset: {}, trying with price oracle", [
```

Comment:
This makes more sense as a warning to me: https://github.com/messari/subgraphs/pull/1960